src/main/java/net/snowflake/ingest/utils/Utils.java

```diff
@@ -430,4 +433,31 @@ public static String concatDotPath(String... path) {
     }
     return sb.toString();
   }
+
+  /**
+   * Get the footer size (metadata size) of a parquet file
+   *
+   * @param bytes the serialized parquet file
+   * @return the footer size
+   */
+  public static long getParquetFooterSize(byte[] bytes) throws IOException {
+    final int magicOffset = bytes.length - ParquetFileWriter.MAGIC.length;
```

Comment:
curious, does this handle both PARE and PAR1 ?
I guess they're both 4 bytes so the length won't change.

Reply:
Yes they have the same length. Added check for `PARE`.